### PR TITLE
New event iron-form-before-submit for change iron-ajax before …

### PR DESCRIPTION
**Pull request by klarkc**
…submit

Now you can change iron-ajax parameters before submit with the iron-form-before-submit event.

Referrer: https://github.com/PolymerElements/iron-form/issues/26#issue-94194486

### CODE REVIEW

**Comment by klarkc** [comment]
I signed it!

File: iron-form.html [hunk 1]
```diff
@@ -143,9 +149,12 @@
       } else {
         this._requestBot.params = json;
       }
-
-      this._requestBot.generateRequest();
-      this.fire('iron-form-submit', json);
+        
+      ev = this.fire('iron-form-before-submit', this._requestBot, {cancelable: true});
```
Comment (notwaldorf) [inline iron-form.html]: `var event = ...` maybe?

File: iron-form.html [hunk 2]
```diff
@@ -132,7 +138,7 @@
       }
 
       var json = this.serialize();
-
+
```
Comment (cdata) [inline iron-form.html]: Please remove the leading whitespace here.

File: test/basic.html [hunk 3]
```diff
@@ -262,13 +262,19 @@
 
     test('can submit with method=post', function(done) {
       form = fixture('FormPost');
-
+      var before_submit = false;
```
Comment (cdata) [inline test/basic.html]: Conventionally, we use `camelCase` for all identifiers. Please change this to use that form.

**Comment by notwaldorf** [comment]
@klarkc Sorry about taking SO long to look at this PR! Would you mind squashing all your commits and adressing the other comments so I can finally merge this? Thanks!

**Comment by notwaldorf** [comment]
@klarkc First of all: I'm SO, so sorry it took so long to get around this PR. There were a bunch of other form problems that came up, and I think this got bumped around a bit.

I talked a little with @cdata about this, and we think it's a bit weird to expose `iron-form` implementation details (the `iron-ajax` object) that are not in the public API. What do you think about making the `requestBot` a public property? That way, in the pre-submit handler, you wouldn't have to pass it as a parameter, since it would already be available.

Do you think this is a change you're interesting in making, or do you want me to pick it up and submit it as a different PR?

**Comment by klarkc** [comment]
So long time has been passed since this PR that I don't remember the things I've done here. Eventually I'll back to this, but for now I am busy in another projects. Fell free to hook up and post another PR, since this can take a long time to be implemented by me (:

**Comment by notwaldorf** [comment]
@klarkc Sounds good. I'll close this and handle it in a different PR. Thanks, and really sorry about the delay again.

**Pull request closed without merging**